test/e2e/kubectl/portforward.go

```diff
@@ -255,6 +274,75 @@ func doTestConnectSendDisconnect(ctx context.Context, bindAddress string, f *fra
 	))
 }
 
+// Reproduces issue #111825 (https://github.com/kubernetes/kubernetes/issues/111825)
+func doTestConnectionReset(ctx context.Context, f *framework.Framework) {
+	ginkgo.By("Creating the target pod")
+	pod := pfResetPod()
+	if _, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, pod, metav1.CreateOptions{}); err != nil {
+		framework.Failf("Couldn't create pod: %v", err)
+	}
+	if err := e2epod.WaitTimeoutForPodReadyInNamespace(ctx, f.ClientSet, pod.Name, f.Namespace.Name, framework.PodStartTimeout); err != nil {
+		framework.Failf("Pod did not start running: %v", err)
+	}
+
+	ginkgo.By("Running 'kubectl port-forward'")
+	func() {
+		cmd := runPortForward(f.Namespace.Name, pod.Name, 8080)
+		defer cmd.Stop()
+
+		ginkgo.By("Attempting connection #1")
+		func() {
+			conn, err := net.Dial("tcp", fmt.Sprintf("127.0.0.1:%d", cmd.port))
+			if err != nil {
+				ginkgo.By("Connection couldn't connect to port")
+				framework.Failf("Connection couldn't connect to port %d: %v", cmd.port, err)
+			}
+			defer func() {
+				ginkgo.By("Closing the connection to the local port")
+				conn.Close()
+			}()
+
+			ginkgo.By("Sending data to connection's local port")
+			if _, err := conn.Write([]byte("test\n")); err != nil {
```

Comment:
add a comment explaining that the image only will send a RST if more than 2 bytes of data are sent, so nobody in the future may try to "optimize" this and find a different behavior